### PR TITLE
[New] Updates to i18 validation and update the i18n files along with the blacklist files

### LINKED ISSUE #1377: Missing translation in admin dash search
**Describe the bug**
go to admin dash then go to 'skill categories table' and try to search the name column.

**To Reproduce**
1. go to admin dash
2. go to 'skill categories table'
3. try to search the name column

**Expected behavior**
translations are missing so they need to be added. right now it just shows the key

**Screenshots**
image

### LINKED ISSUE #1390: Change "Connections" to "My Circle"   FR "Mon Cercle"
Update the Dropdown 
Replace "Connection" with "My Circle" / "Mon Cercle"
image

Update the Results Cards 
Replace " Add Connection" to "Add to My Circle"  / Ajouter à mon cercle

image

Update the Profile Page Heading and Sub-heading
Replace "Private Group" with "My circle"
Replace "Connections" with "Added to My Circle" / Ajouté à mon cercle
image

Update Nav Bar
Replace "Private Group" with "My Circle"
Replace "Connections" with "Added to My Circle" / Ajouté à mon cercle
image

Update "Review Visibility Settings" Modal
image

### LINKED ISSUE #1431: Update language i18n file
**Describe the bug**
The excel file has been provided by casey to update.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### LINKED ISSUE #1449: i18n validation - add a check to see missing words
**Describe the bug**
This would be important when going to production so we dont have missing keys. This validation will be used to see if there are any words used in the program that are not in the i18n file 

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by mo-5**
#### ⭐ Changes introduced

- Added new validation to see what i18 keys are used in the application but not in translation to prevent having "visibility.selector" something like this in production 
- Cleaned up the missing keys in the admin page 
- Fixed a few edge cases in the validation 
- Before merging please confirm the translation of the following: "visibility.selector": "Select visibility",  "visibility.selector": "Sélectionnez la visibilité",

#### 🔗 Related issue(s)

closes #1449, closes #1377

#### ☑️ Checklist
If the database has been modified:
- [ ] Update database diagram 
- [ ] Create new migration 

If API endpoints has been modified:
- [ ] Update backend documentation 
- [ ] Create/Update validators for the API endpoints 
- [ ] The API endpoints are properly secured with keycloak 

If the UI has been modified:
- [ ] It is translated in both language (with no hard coded text) 
- [ ] The modifications are tab friendly
- [ ] It is accessible

If translations have been modified:
- [ ] run `yarn i18n:validate" and clear errors

### CODE REVIEW

File: services/frontend/src/i18n/tools/validation.js [hunk 1]
```diff
@@ -26,21 +28,44 @@ const blacklistedKeys = require("./blacklistKeys.json");
 (async () => {
   console.log("\n************ Starting i18n Validator ****************\n");
 
+  // Remove all blacklisted key from the check
```
Comment (dynamic11) [inline services/frontend/src/i18n/tools/validation.js:30]: repeat

File: services/frontend/src/i18n/tools/validationHelperFunctions.js [hunk 2]
```diff
@@ -148,84 +148,120 @@ const getFilesInDirectory = async (dir, ext) => {
...
   let files = await Promise.all(
     ext.map((extension) => getFilesInDirectory(dir, extension))
   );
+
   files = _.flatten(files);
   const filesContent = await Promise.all(
     files.map(async (i) => fs.readFile(i).then((buffer) => buffer.toString()))
   );
 
-  // function to search for key in file content
```
Comment (dynamic11) [inline services/frontend/src/i18n/tools/validationHelperFunctions.js:168]: please move this comment to change

File: services/frontend/src/i18n/tools/validationHelperFunctions.js [hunk 3]
```diff
@@ -148,84 +148,120 @@ const getFilesInDirectory = async (dir, ext) => {
...
- * A test to fined unused translations in the entire project
- *
- * @param {object} enTranslations List of english translations
- * @param {object} frTranslations List of english translations
- * @param {string[]} blacklistedKeys List of keys to be ignored
- */
-const findUnusedTranslations = async (
-  enTranslations,
-  frTranslations,
+const findMissingTranslations = async (
```
Comment (dynamic11) [inline services/frontend/src/i18n/tools/validationHelperFunctions.js]: please add comment to function

File: services/frontend/src/i18n/tools/validationHelperFunctions.js [hunk 4]
```diff
@@ -148,84 +148,120 @@ const getFilesInDirectory = async (dir, ext) => {
...
+  const getValue = (data) =>
+    _.compact(
+      data.map((i) => {
+        const value = i.replace(/,|\s/g, "");
+        if (value.match(/["|`|'].*['|"|`]/)) {
+          const cleanedValue = value
+            .match(/["|`|'].*['|"|`]/)[0]
+            .replace(/['|"|`]+/g, "");
+
+          if (
```
Comment (dynamic11) [inline services/frontend/src/i18n/tools/validationHelperFunctions.js]: maybe we should move this into a separate function. let me know what you think :)

**Review by dynamic11 — CHANGES_REQUESTED** [review]
please see comments. Also please remove mena's translations into a separate PR

**Review by Rizvida1 — APPROVED** [review]

File: services/frontend/src/i18n/tools/validationHelperFunctions.js [hunk 5]
```diff
@@ -139,80 +134,49 @@ const getFilesInDirectory = async (dir, ext) => {
...
- * @param {string} dir Directory to search files
- * @param {string[]} ext Extensions of the files to be searched
- * @param {string[]} searchableKeys  Keys to search for in the directory
- * @param {string[]} ignoreKeys  Keys that should not be returned even
- *                               if they were not found in directory
- * @returns {string[]}  Returns keys that were not used in the project
+ * returns the content of a file in a string format
+ * @param {string} dir
+ * @param {string[]} ext
+ * @returns {string[]}
```
Comment (dynamic11) [inline services/frontend/src/i18n/tools/validationHelperFunctions.js]: better comments needed

**Review by dynamic11 — APPROVED** [review]
approved but please see minor comment

**Pull request merged**